runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2970,6 +2976,15 @@ void TR::CompilationInfo::prepareForRestore()
 
    if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerboseCheckpointRestore))
       TR_VerboseLog::writeLineLocked(TR_Vlog_CHECKPOINT_RESTORE, "Ready for restore");
+
+   // Check if there is no swap memory post restore
+   J9MemoryInfo memInfo;
+   if ((omrsysinfo_get_memory_info(&memInfo) == 0) && (0 == memInfo.totalSwap))
+      {
+      if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerboseCodeCache))
+         TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "At Checkpoint Restore:: Swap Memory disabled");
+      compInfo->setIsSwapMemoryDisabled();
```

Comment:
If during the checkpoint we find the JVM to not use swap, but after the restore the JVM does use swap, we must revert that decision. Thus, we must use something like `setIsSwapMemoryIsDisabled(false)`. Again, you don't need to use `compInfo` because you are already in an instance of `compInfo`.